### PR TITLE
New max and min time API

### PR DESCRIPTION
**Pull request by tulioag**

### CODE REVIEW

File: vaadin-time-picker-flow/src/main/java/com/vaadin/flow/component/timepicker/TimePicker.java [hunk 1]
```diff
@@ -400,52 +400,112 @@ public Locale getLocale() {
...
+     * Sets the minimum time in the time picker. Times before that will be
+     * disabled in the popup.
+     *
+     * @param min
+     *            the minimum time that is allowed to be selected, or
+     *            <code>null</code> to remove any minimum constraints
+     */
+    public void setMinTime(LocalTime min) {
+        this.min = min;
+        super.setMin(min.format(initializeAndReturnFormatter()));
```
Comment (DiegoCardoso) [inline vaadin-time-picker-flow/src/main/java/com/vaadin/flow/component/timepicker/TimePicker.java]: This is going to throw a `NPE`, right?

File: vaadin-time-picker-flow/src/main/java/com/vaadin/flow/component/timepicker/TimePicker.java [hunk 2]
```diff
@@ -400,52 +400,112 @@ public Locale getLocale() {
...
+     * Sets the maximum time in the time picker. Times after that will be
+     * disabled in the popup.
+     *
+     * @param max
+     *            the maximum time that is allowed to be selected, or
+     *            <code>null</code> to remove any maximum constraints
+     */
+    public void setMaxTime(LocalTime max) {
+        this.max = max;
+        super.setMax(max.format(initializeAndReturnFormatter()));
```
Comment (DiegoCardoso) [inline vaadin-time-picker-flow/src/main/java/com/vaadin/flow/component/timepicker/TimePicker.java]: `NPE`?

**Review by DiegoCardoso — DISMISSED** [review]
It needs test cases for `null` for the new API and how it's work with the deprecated one.

File: vaadin-time-picker-flow/src/main/java/com/vaadin/flow/component/timepicker/TimePicker.java (hunk 1 above)
Comment (tulioag) [reply vaadin-time-picker-flow/src/main/java/com/vaadin/flow/component/timepicker/TimePicker.java]: Fixed

File: vaadin-time-picker-flow/src/main/java/com/vaadin/flow/component/timepicker/TimePicker.java (hunk 2 above)
Comment (tulioag) [reply vaadin-time-picker-flow/src/main/java/com/vaadin/flow/component/timepicker/TimePicker.java]: Fixed

**Review by Haprog — APPROVED** [review]
Looks good.
